pkg/term/console_windows.go

```diff
@@ -85,3 +344,762 @@ func GetConsoleScreenBufferInfo(fileDesc uintptr) (*CONSOLE_SCREEN_BUFFER_INFO,
+				term.inputEscapeSequence = []byte(KEY_ESC_CSI)
+			default:
+			}
+		}
+	case "h":
+		for _, value := range parsedCommand.Parameters {
+			switch value {
+			case "?25", "25":
+				SetCursorVisible(uintptr(handle), BOOL(1))
+			case "?1049", "1049":
+				// TODO (azlinux): Save terminal
+			case "?1", "1":
+				// If the DECCKM function is set, then the arrow keys send application sequences to the host.
+				// DECCKM (default off): When set, the cursor keys send an ESC O prefix, rather than ESC [.
+				term.inputEscapeSequence = []byte(KEY_ESC_O)
+			default:
+			}
+		}
+
+	case "]":
+	/*
+		TODO (azlinux):
+			Linux Console Private CSI Sequences
+
+		       The following sequences are neither ECMA-48 nor native VT102.  They are
+		       native  to the Linux console driver.  Colors are in SGR parameters: 0 =
+		       black, 1 = red, 2 = green, 3 = brown, 4 = blue, 5 = magenta, 6 =  cyan,
+		       7 = white.
+
+		       ESC [ 1 ; n ]       Set color n as the underline color
+		       ESC [ 2 ; n ]       Set color n as the dim color
+		       ESC [ 8 ]           Make the current color pair the default attributes.
+		       ESC [ 9 ; n ]       Set screen blank timeout to n minutes.
+		       ESC [ 10 ; n ]      Set bell frequency in Hz.
+		       ESC [ 11 ; n ]      Set bell duration in msec.
+		       ESC [ 12 ; n ]      Bring specified console to the front.
+		       ESC [ 13 ]          Unblank the screen.
+		       ESC [ 14 ; n ]      Set the VESA powerdown interval in minutes.
+
+	*/
+	default:
+	}
+	return len(command), nil
+}
+
+// WriteChars writes the bytes to given writer.
+func (term *WindowsTerminal) WriteChars(w io.Writer, p []byte) (n int, err error) {
+	return w.Write(p)
+}
+
+const (
+	CAPSLOCK_ON        = 0x0080 //The CAPS LOCK light is on.
+	ENHANCED_KEY       = 0x0100 //The key is enhanced.
+	LEFT_ALT_PRESSED   = 0x0002 //The left ALT key is pressed.
+	LEFT_CTRL_PRESSED  = 0x0008 //The left CTRL key is pressed.
+	NUMLOCK_ON         = 0x0020 //The NUM LOCK light is on.
+	RIGHT_ALT_PRESSED  = 0x0001 //The right ALT key is pressed.
+	RIGHT_CTRL_PRESSED = 0x0004 //The right CTRL key is pressed.
+	SCROLLLOCK_ON      = 0x0040 //The SCROLL LOCK light is on.
+	SHIFT_PRESSED      = 0x0010 // The SHIFT key is pressed.
+)
+
+const (
+	KEY_CONTROL_PARAM_2 = ";2"
+	KEY_CONTROL_PARAM_3 = ";3"
+	KEY_CONTROL_PARAM_4 = ";4"
+	KEY_CONTROL_PARAM_5 = ";5"
+	KEY_CONTROL_PARAM_6 = ";6"
+	KEY_CONTROL_PARAM_7 = ";7"
+	KEY_CONTROL_PARAM_8 = ";8"
+	KEY_ESC_CSI         = "\x1B["
+	KEY_ESC_N           = "\x1BN"
+	KEY_ESC_O           = "\x1BO"
+)
+
+var keyMapPrefix = map[WORD]string{
+	VK_UP:     "\x1B[%sA",
+	VK_DOWN:   "\x1B[%sB",
+	VK_RIGHT:  "\x1B[%sC",
+	VK_LEFT:   "\x1B[%sD",
+	VK_HOME:   "\x1B[1%s~", // showkey shows ^[[1
+	VK_END:    "\x1B[4%s~", // showkey shows ^[[4
+	VK_INSERT: "\x1B[2%s~",
+	VK_DELETE: "\x1B[3%s~",
+	VK_PRIOR:  "\x1B[5%s~",
+	VK_NEXT:   "\x1B[6%s~",
```

Comment:
The keys are supposed to be translated to a specific escape sequences.
If any of the keys is not working as expected then perhaps mapping is wrong

As for the specific translation of keystrokes for arrows - the vi specifically asks for sending application mapping and that is what we send.
Can I take this as a specific bug and fix it later in a smaller PR for bug fixes?